pkg/scheduler/backend/queue/scheduling_queue.go

```diff
@@ -512,15 +527,18 @@ func (p *PriorityQueue) runPreEnqueuePlugins(ctx context.Context, pInfo *framewo
 			continue
 		}
 		pInfo.UnschedulablePlugins.Insert(pl.Name())
+		pInfo.GatingPlugin = pl.Name()
+		pInfo.Gated = true
```

Comment:
IMHO it was better when pInfo.Gated was set explicitly in one place. Note that the gatedBefore was captured prior changing the value.

Alternatively consider returning just the gatedPlugin name which would indicate whether the pod was gated or not

pInfo.GatingPlugin = runPreEnqueuePlugins(...)
pInfo.Gated = pInfo.GatingPlugin != ""

Actually, cannot pInfo.GatedPlugin replace pInfo.Gated? or make pInfo.Gated a function that returns pInfo.GatedPlugin != ""

Reply:
> Actually, cannot pInfo.GatedPlugin replace pInfo.Gated?

We do have to know which plugin is gating this pod, so the answer is no.

> make pInfo.Gated a function that returns pInfo.GatedPlugin != ""

But, this is a good one.